File: WorkIt/app/src/main/java/prot3ct/workit/views/login/base/LoginContract.java
package prot3ct.workit.views.login.base;

import prot3ct.workit.base.BaseView;

public interface LoginContract {
    interface View extends BaseView<Presenter> {
        void showListJobsActivity();

        void showRegisterActivity();

        void notifyError(String errorMessage);

        void showDialogforLoading();

        void dismissDialog();
    }

    interface Presenter {
        void loginUser(String email, String password);

        void autoLoginUser();

        boolean isUserLoggedIn();
    }
}

File: WorkIt/app/src/main/java/prot3ct/workit/views/login/base/LoginContract.kt
package prot3ct.workit.views.login.base

import prot3ct.workit.base.BaseView

interface LoginContract {
    interface View : BaseView<Presenter> {
        fun showListJobsActivity()
        fun showRegisterActivity()
        fun notifyError(errorMessage: String)
        fun showDialogforLoading()
        fun dismissDialog()
    }

    interface Presenter {
        fun loginUser(email: String, password: String)
        fun autoLoginUser()
        val isUserLoggedIn: Boolean
    }
}